src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -306,6 +306,12 @@ export class MdAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
    */
   private _setValueAndClose(event: MdOptionSelectionChange | null): void {
     if (event) {
+      this.autocomplete.options.forEach((option) => {
+        if (option != event.source) {
+          option.deselect();
```

Comment:
Done, but instead of setting a filter on `optionSelections`, I've change to only call the deselect function on the option that was previously selected.

Please let me know if this PR needs any other changes